wandb/sdk/internal/staging.py

```diff
@@ -0,0 +1,44 @@
+import os
+from pathlib import Path
+from typing import Optional, Union
+
+from wandb import env
+from wandb.errors import term
+from wandb.sdk.lib import filesystem
+
+_staging_dir: Optional[Path] = None
+
+
+def get_staging_dir() -> Path:
+    """Return the staging directory for the current run."""
+    global _staging_dir
+    if _staging_dir is None:
+        path = Path(env.get_data_dir()) / "artifacts" / "staging"
+        filesystem.mkdir_exists_ok(path)
+        _staging_dir = path.expanduser().resolve()
+    return _staging_dir
+
+
+def is_staged_copy(local_path: Union[str, os.PathLike]) -> bool:
+    """Returns True if the given path is a staging copy of a local file."""
+    try:
+        # Raises if the path is not a child of the staging directory.
+        Path(local_path).relative_to(get_staging_dir())
+        return True
+    except ValueError:
+        return False
```

Comment:
```suggestion
    return Path(local_path).is_relative_to(get_staging_dir())
```

Reply:
I know, I had it that way originally, but `is_relative_to` was only introduced in Python 3.9 😕